src/lib.rs

```diff
@@ -68,18 +66,20 @@ fn generate_profile(document:String) -> Vec<(String)> {
     profile_vec.sort_by(|a,b| b.1.cmp(&a.1));
 
     //remove the frequency count from the vector.
-    let mut profile_vec_no_count = Vec::new();
+    let mut top_frequent_grams = Vec::new();
     let mut index:usize = 0;
     while index < MAX_PROFILE_LEN  && index < profile_vec.len() {
-        profile_vec_no_count.push(profile_vec[index].0.clone());
+        top_frequent_grams.push(profile_vec[index].0.clone());
         index += 1;
     }
-    profile_vec_no_count
+    top_frequent_grams
 
```

Comment:
I believe this section can be shortened and made slightly more readable using iterators. I haven't tested the code bellow, but it should give you a good starting point.
```rust
profile_vec.into_iter().take(MAX_PROFILE_LEN).map(|tuple| tuple.0.clone()).collect()
```